src/wp-includes/post.php

```diff
@@ -5751,39 +5751,45 @@ function get_page_by_path( $page_path, $output = OBJECT, $post_type = 'page' ) {
  * @return WP_Post|array|null WP_Post (or array) on success, or null on failure.
  */
 function get_page_by_title( $page_title, $output = OBJECT, $post_type = 'page' ) {
-	global $wpdb;
 
-	if ( is_array( $post_type ) ) {
-		$post_type           = esc_sql( $post_type );
-		$post_type_in_string = "'" . implode( "','", $post_type ) . "'";
-		$sql                 = $wpdb->prepare(
-			"
-			SELECT ID
-			FROM $wpdb->posts
-			WHERE post_title = %s
-			AND post_type IN ($post_type_in_string)
-		",
-			$page_title
-		);
-	} else {
-		$sql = $wpdb->prepare(
-			"
-			SELECT ID
-			FROM $wpdb->posts
-			WHERE post_title = %s
-			AND post_type = %s
-		",
-			$page_title,
-			$post_type
-		);
+	$last_changed = wp_cache_get_last_changed( 'posts' );
+
+	$hash      = md5( $page_title . serialize( $post_type ) );
```

Comment:
The hash will end up different if the post type is passed as `'page'` versus `['page']`. It would be good to cast to an array before generating the hash.

```suggestion
	$post_type = (array) $post_type;
	$hash      = md5( $page_title . serialize( $post_type ) );
```